Hrm running with `pytest -n auto` on my local machine, I saw this just now:

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/_pytest/main.py", line 193, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/_pytest/main.py", line 237, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/pluggy/manager.py", line 92, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/pluggy/manager.py", line 86, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/xdist/dsession.py", line 112, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/xdist/dsession.py", line 135, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/usr/local/google/home/mattjj/miniconda3/lib/python3.7/site-packages/xdist/dsession.py", line 177, in worker_workerfinished
INTERNALERROR>     assert not crashitem, (crashitem, node)
INTERNALERROR> AssertionError: ('tests/loops_test.py::LoopsTest::test_error_early_exit_range', <WorkerController gw5>)
INTERNALERROR> assert not 'tests/loops_test.py::LoopsTest::test_error_early_exit_range'
```

Not sure how that came out as an internal error, but the test case looks suspicious. I wonder if something is going on with threading...

--- a/jax/core.py
+++ b/jax/core.py
@@ -544,11 +544,10 @@ trace_state = TraceState()
 
 def reset_trace_state() -> bool:
   "Reset the global trace state and return True if it was already clean."
-  global trace_state
   if (trace_state.substack != [Sublevel(0)] or
       trace_state.trace_stack.downward or
       trace_state.trace_stack.upward):
-    trace_state = TraceState()
+    trace_state.__init__()
     return False
   else:
     return True

--- a/jax/core.py
+++ b/jax/core.py
@@ -547,7 +547,7 @@ def reset_trace_state() -> bool:
   if (trace_state.substack != [Sublevel(0)] or
       trace_state.trace_stack.downward or
       trace_state.trace_stack.upward):
-    trace_state.__init__()
+    trace_state.__init__()  # type: ignore
     return False
   else:
     return True

